website/common/script/ops/buy/buyQuestGold.js

```diff
@@ -53,7 +46,8 @@ export class BuyQuestWithGoldOperation extends AbstractGoldItemOperation { // es
 
   checkPrerequisites (user, questKey) {
     const item = content.quests[questKey];
-    if (questKey === 'lostMasterclasser1' && !this.userAbleToStartMasterClasser(user)) {
+    if (questKey.startsWith('lostMasterclasser')
+    && !this.userAbleToStartMasterClasser(user)) {
```

Comment:
This is the cause of one of your test failures. You removed the `userAbleToStartMasterClasser` function but still try to invoke it here.

Reply:
putting it back in--there should be a way to call `locked` into `checkPrerequisites` but I think that's going to be part of the server-side fix when it's implemented